### PR TITLE
[FLINK-28485][doc] Structure examples better and add README/doc

### PR DESCRIPTION
**Pull request by pvary**
## What is the purpose of the change

Create a README.md file for the examples. Describing the usage of those.

## Brief change log

Some descriptions, and usage examples

## Verifying this change

This change is a documentation change without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changes to the `CustomResourceDescriptors`: no
  - Core observer or reconciler logic that is regularly executed: no

## Documentation

  - Does this pull request introduce a new feature? no

### CODE REVIEW

File: examples/README.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+This is a simple deployment defined by a minimal deployment file.
+The configuration contains the following:
+- Defines the job to run
+- Assigns the resources available for the job
+- Defines the parallelism used
+
+To run the job submit the yaml file using kubectl:
+```bash
+cd examples
```
Comment (mbalassi) [inline examples/README.md:45]: Since this Readme is in `examples` you can omit this imho.

**Review by mbalassi — COMMENTED** [review]
Thanks @pvary, looks good - suggested minor improvements.

File: examples/README.md [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+# Flink Kubernetes Operator Examples
+
+## Overview
+
+This directory contains few examples for Flink Kubernetes Operator.
```
[the remark this answers is not in the document]
Comment (pvary) [reply examples/README.md]: Fixed.
Thanks!

File: examples/README.md [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+These examples should only serve as starting points when familiarizing yourself with the
+Flink Kubernetes Operator and users are expected to extend these based on their production needs.
+
+## Usage
+
+Please refer to the [Quick Start](https://nightlies.apache.org/flink/flink-kubernetes-operator-docs-main/docs/try-flink-kubernetes-operator/quick-start)
+to run the examples.
```
[the remark this answers is not in the document]
Comment (pvary) [reply examples/README.md]: Fixed

File: examples/README.md (hunk 1 above)
Comment (pvary) [reply examples/README.md]: Removed

File: examples/README.md [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+
+To try out this run the following command:
+```bash
+cd examples
+kubectl apply -f basic-session-deployment-and-job.yaml
+```
+
+### SQL runner
+
+For running Flink SQL scripts check this [example](https://github.com/apache/flink-kubernetes-operator/tree/main/examples/flink-sql-runner-example).
```
[the remark this answers is not in the document]
Comment (pvary) [reply examples/README.md]: Done, and fixed when used absolute link to the `kustomize` example as well

File: examples/README.md [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+
+To try out this run the following command:
+```bash
+kubectl apply -f basic-session-deployment-and-job.yaml
+```
+
+### SQL runner
+
+For running Flink SQL scripts check this [example](flink-sql-runner-example/README.md).
+
```
Comment (bgeng777) [inline examples/README.md:97]: Hi @pvary, the python example has been merged as well and it would be great if we can add it in this doc like SQL runner.

Comment (pvary) [reply examples/README.md:97]: Thanks for the heads up @bgeng777!
Added the link to the python example

File: examples/README.md [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+```
+
+#### High availability
+
+Basic example to configure Flink Deployments in
+[HA mode](https://nightlies.apache.org/flink/flink-docs-release-1.15/docs/deployment/ha/overview/).
+The example shows how to set savepoint directory, checkpoint directory and HA. To try out this run the following command:
+```bash
+kubectl apply -f basic-checkpoint-ha.yaml
+```
```
Comment (gyfora) [inline examples/README.md:152]: I would not call this "advanced" as this is a very general requirement. We should have this near the top and also I would prefer to call it `Checkpointing & High Availability`

**Review by gyfora — CHANGES_REQUESTED** [review]
Overall looks really good, thank you!

Added a minor request, happy to merge afterwards

File: examples/README.md (hunk 6 above)
Comment (pvary) [reply examples/README.md]: Thanks for the insight!
Done

**Comment by pvary** [comment]
Thanks for the review @mbalassi, @bgeng777 and @gyfora for the merge!

**Pull request merged**